doc/rtd/applications.rst

```diff
@@ -666,3 +666,124 @@ regular expression.
+
+--ivt IVT_VARIABLE
+    name of variable with integrated vapor transport (IVT).
+
+--n_threads N_THREADS
+    number of threads used in PyTorch for parallelizing intra CPU operations.
+
+--binary_ar_threshold THRESHOLD
+    probability threshold for segmenting AR probabilities.
+
+--filename_time_template TEMPLATE
+    An optional way to read time template information from input filenames
+
+--compression_level COMPRESSION_LEVEL
+    the compression level used for each variable.
+
+--date_format DATE_FORMAT
+    the format for the date to write in the filename
+
+--first_step FIRST_STEP
+    first time step to process.
+
+--last_step LAST_STEP
+    last time step to process.
+
+--steps_per_file STEPS_PER_FILE
+    number of time steps to write to each output file.
+
+--start_date START_DATE
+    an optional way to further specify the time range to process. The accepted
+    format is a CF style human readable date spec such as YYYY-MM-DD hh:mm:ss.
+    Because of the space in between day and hour spec quotes must be used. For
+    example "2005-01-01 00:00:00". Specifying a start date is optional, if
+    none is given then all of the time steps in all of the files specified in
+    the `--input_regex` are processed.
+
+--end_date END_DATE
+    see `--start_date`. this is has a similar purpose in restricting
+    the range of time steps processed.
+
+--periodic_in_x FLAG
+    Flags whether the x dimension (typically longitude) is periodic.
+
+Example
+~~~~~~~
+.. code-block:: bash
+
+    teca_bayesian_ar_detect --n_threads 1                            \
+        --input_regex "${data_dir}/ARTMIP_MERRA_2D_2017-05.*\.nc$"   \
+        --output_file test_deeplabv3p_ar_detect_app_output_%t%.nc
+
+Parallel Example
+~~~~~~~~~~~~~~~~
+
+The following example documents SLURM script that was used to generate output
+used by O'Brien et al. (see citation above).  This used 1520 nodes, and
+simultaneously ran 1,024 AR detectors on the 37 years of the MERRA-2 reanalysis
+in approximately 2 minutes on the Cori KNL supercomputer at NERSC:
+
+.. code-block:: bash
+
+   #!/bin/bash
+  #SBATCH -J bard_merra2
+  #SBATCH -N 1520
+  #SBATCH -C knl
+  #SBATCH -q regular
+  #SBATCH -t 00:20:00
+
+  WORKDIR=$SCRATCH/teca_bard_merra2_artmip
+  mkdir -p ${WORKDIR}
+  cd $WORKDIR
+
+  module load teca/develop_haswell
+  HDF5_USE_FILE_LOCKING=FALSE
+
+  for year in `seq 1980 2017`
+  do
+      echo "Starting ${year}"
+      srun -n 680 -c 16 -N 40 --cpu_bind=cores teca_bayesian_ar_detect \
+          --input_regex "/global/project/projectdirs/m1517/cascade/external_datasets/ARTMIP/MERRA_2D/${year}/ARTMIP_MERRA_2D_.*\.nc" \
+          --cf_reader::t_axis_variable "" \
+          --cf_reader::filename_time_template  "ARTMIP_MERRA_2D_%Y%m%d_%H.nc" \
+          --steps_per_file 3000 \
+          --cf_writer::date_format "%Y" \
+          --output_file MERRA2.ar_tag.teca_bard_v1.0.3hourly.%t%.nc4 &> bard_${year}_${SLURM_JOB_ID}.log &
+  done
```

Comment:
This is a great example, and shows using the filename_time_template option. We should keep this example as is, and add an additional example of processing HighResMIP data.